docs/assets/diagrams/MIE-architecture.xml

```diff
@@ -1 +1 @@
3/MPuOHJd2XH1xL+jaj3XW18wiSkRNiRevJSp4sytTJXs7wtXrQ1+qavnBZqnV2qIa9Dth9ftUUnsaGxUsVtB28awgXV63e67J1Ut1gjYVlOz2Ha4j2bJqQIo3rGkTgawBOV5PBHdN3EivrlryPbwisruOvtIxdAbRRWeAiMnnP97xwHbeiMc8F6pBebAk6pJ0FSR0N2iIU/IOann+sI6S+NS/Ou2DPXhjP65WS4mzIIZYn5HiT9Tot/dS5Ibjfa5ShJEH1bTtpqeV+HxOsFzSgTEblFuO6DAY0874PIIuMSC+HdzRvoJJ71R3ze0lBigf3J0qkHAzxkmqCj343yMxwQR1eO4Ux5Wl3w6NTnz9SXXeUE3/WM72jQs6ITGrJmTmJNNf5I0YbhJA3viTxLmQgdDx+uXoeOo/qTLZc4EAQIhw6qRsy1pMs+qOeY+fmLM3z019kjSpsZwumpTczsu3DfiOHf8QyOdX/oljcsNCTJ7sLWbQuTEiFtUQsmvanFHDuMlIrRurfARzccRGiyxI1hiSUVqo75YT8kC7GZ3GctRtVznjaiK4DiqwrGaqsJvs8a4qtiQJTJEb95a9EbNa++sM8yt1VxLgaCCuL1XZvVgyLogyc4Ls2r9pQBpLeTMMdSxnJHleS6wQ+DB7ayFGtaG5jh7moD/U1z8XfvMI/28tVqWsUZjVVa6qEegHcjzvX4s6dyucZ+wZ2EfMfBBwPcQcC8YeesFXOOHObKI62rrDLztxlsHbOLtiTnrqVIrqgghljcMPuECRWK7WUpYkakXcf5fs1tZlAcTrNYU1YBCjBquVN8cszWbZrIsoTTiMUWQ3jOWf7g+Y2W/2NNMqhIb9NM9KZ1zZZWrDVsReTGs+yWd0jEpYYSS5+rcElE31XFC2XBepqaDhzhfUrb/rEYxkuUWiFKAQfZ8neFnOUaU5Cn9s8zZyFAUlUAoI1McuLkALsgxeYgnmA3mFzya0u8C1wnKKUxyjMhkVpWBkGuj33FWLNn+fNaLi8mvjADsu+7paiCJi3J/ByNJTOQAJijHo77LyNoFlDxwrLhD2PKiabZ+28LN3RATY2WpZOrN4Fh72441Jclxh5C7uQ3g6jp5VXbgj9sBfMfI+/hhNtnDVjbU9gF5Mm1q0HtvWu85sI/og4GuGOT+9VupOcqMzUcSlzbanC9q0vE3ilVpjNJ+sbzpWLn1dJ4uMPKFe+uf2iG9q+vVZJi33FddchmqwOfBozShyJGQtcMMRmnQnx+/eufn5Lfb53dZ9P7n4tu5sxb/Sm05LhBqyTt24DwvGcpq3Vkge1JFYxXxaS+r5CpuY9niZrW7lkBQJVQ08VsvkMugyRCec8x18dFIXZ4xype8cC/O3Cv92khbJ7Ypg2sUZnd8b3DUuyNQ/2kiyrd1rjxfo2ftg6hZDcjUnIFaoevPdAWMKQGRtoTLW6hfZoLplhc2Z1PdEtbTMdlYlR5ZdH57ZWWFBuYqKMNAQxlPZ2a45igDXkH95A113ME5/f+PJS4JHa+Oc+9cpKlM+jaqkJou6t6xvvgWcdtuhJ8S+od6Q8JQa/et1drdAcCy1m7QVO6akp2iLknDb2tuBQnUSW+IZR2C1/KlA2IzuFpUmXaqvLYcxxCvrWBDHcKhOmv/qrPuDjoHBO0sNkuB3ZE9pBoflURdK44IbqtQ4wDFo2yIq9H+jgd5zU5/ytqmY+Vgc7XIRZJIP7OyBYm3ewYPv51+3QaOtoKxrNZNTOdl64J9Q8Sl5xGXPZSFr5RyPvlUtqnQ+LAloN9bAvaw4MP98GdsPwAU1Xx1G0BfzXudDKVR7c5fG7RL09pANUigSACs89hYuiEUeVTHNUgakU+pc1ZhT3DW3T9zcOMGAs6l7VHOnlUVkiMfSrz2YdbYr8TrPqvwhlZxVc/PsecNNUnpVTmxywBOFBMmCRD84JGcMqmYuyaz1QsOJ6uJcHB+99n5rXk17LG938FJNmWvkpNC+2yHnXi00d5Lt66AnL+tgJzZGV+Un9nqtwhgx6ldbg3py9QuXnX2aqHz+hDgHhoB5aWUBNWOGH4Cz5ha6wKzHK+pyOyAvwfzZocrQAuB1SgO6ykTpN2U0KKU/ykfH5PrB1zljVmVxRWVMAFtG7GxhtAkAxx6elLmv7YxOM8nCI/KoxmJczx6xGN6c3w3xQsqABPV+PWv2L+OM/dL5WGH6dK2vPbyNPDVKdPVmJLGQojQVyH1lSUvtmGlZjRq4dHFwbDWTaCzypoKdj1HG5tnj8XRdvzDAbbGBNLw01x6i68m0A4L1N4uUHeHnOu3VIjn2yPbtqFrha5twVOXOIG+xkMyRHN6E83ZHXFe2C6ksyfmzK36/NOY7kd520pXu1tq/u12t39wu/uF7NvwCsEhCteXKNzumsMHzeoPtmv3bK46SdGml8h418CJlKneyLgmaWyo6d/jmv67i3sAnX6bpuLrTmkmOE0zwXH3enXScVRI2C8VIhzGfQ56XRJclsNhZ/yZRVVpnCrv/gonuGzWKnvzQ2L34fo9p681mtWTDUt7rLNFLX9pu1iq/nJ0ZcqkhBmQAHVt/ZLqAEqE9voJT5YlRlf7YfcPuQ4uQx2uWvtkZeSgjitZRrKZNG0IVsGG5dfgMnz9LkMZbpcJ3mFw8pcqBZpi3IOb8BW7CW3YTr7cBWfmUrzFVvd/s28w6PqGnODg7wJ4WRk4zYvXBsfg63cM2rBZEsYC7ulnJAuoO2Z77Q0Uwrq9SDrol1TDNVP/4Ap8K65Ap7V67oXJaVmneUdDr/x/3ZUG7JXSsDRBwt55/7aVLKpVGevgKBxcgS93Bc7TB16pq12Er2ynBSpq7YhRvtam7Kw3m6W/Nk7eL0gis4CmQJjW02gfRlVqCuSpi3MFd7XCaZ2rpGlItQNdXFfjKZPTReNNEYcpi64hzIa61IOrrA+usq21DdehvfPmr641LvYQTtokKZtR5LH37Gk/pVQx0c7/Aw==</diagram></mxfile>
```

Comment:
I replaced the Elasticsearch objects with abstract references to data stream consumers.